### PR TITLE
Bugfix FXIOS-6750 [v124] Share button in URL bar color after 1 click wrong color

### PR DESCRIPTION
**Pull request by rajparad**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-6750)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/15023)

## :bulb: Description

After debugging I found that in `ShareButton` class,
`override open var isHighlighted: Bool { didSet { self.tintColor = isHighlighted ? selectedTintColor : unselectedTintColor } }` seems the cause of issue. once the button is pressed isHighlighted gets triggered. based on condition it is changing colour to selectedTintColor.

- With didSet we are changing image tint whenever event occurs with Button. after changing the colour we are not reassigning the old colour.
- once we remove/change this condition above  self.tintColor = isHighlighted ? unselectedTintColor : selectedTintColor, it's working  as per expected behaviour.

https://github.com/mozilla-mobile/firefox-ios/assets/59967463/488885fa-8f29-4e0c-80b7-8e28644a42c7

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by lmarceau** [comment]
Heya! Thanks for the PR @rajparad. I hadn't understood the `isHighlighted` property would be removed entirely, but rather modified. We want to keep a highlighted color. I think we should aim to have something similar to the [`ReaderModeButton` implementation](https://github.com/mozilla-mobile/firefox-ios/blob/fcdb60fa81d4f8f031ffbc39e014859bbfbcc975/firefox-ios/Client/Frontend/Toolbar%2BURLBar/ReaderModeButton.swift#L36-L40) instead. Could you take a look? Thank you! 🙏

**Review by lmarceau — CHANGES_REQUESTED** [review]
As commented

**Comment by rajparad** [comment]
Hello @lmarceau , Thanks for the example. 

Sorry I got bit confused. I found that `override open var isHighlighted` works fine, but the colour that has not been assigned properly based on name it refers. the **var** had `selectedTintColor = theme.colors.iconSecondary` and `unselectedTintColor = theme.colors.iconDisabled` values. Based on the name of variable and selectedColor which refers to have dimmed/disabled tint should have `theme.colors.iconDisabled` value and same for `unselectedColor`. 

After checking in few parts of project and changing value accordingly, it now works fine :). Let me know if you have questions.

File: firefox-ios/Client/Frontend/Toolbar+URLBar/ShareButton.swift [hunk 1]
```diff
@@ -14,10 +14,10 @@ class ShareButton: UIButton {
     var disabledTintColor: UIColor!
 
     override open var isHighlighted: Bool {
-        didSet {
-            self.tintColor = isHighlighted ? selectedTintColor : unselectedTintColor
+            didSet {
+                self.tintColor = isHighlighted ? selectedTintColor : unselectedTintColor
+            }
```
Comment (lmarceau) [inline firefox-ios/Client/Frontend/Toolbar+URLBar/ShareButton.swift:19]: It seems this is possibly too much indented?

Comment (rajparad) [reply firefox-ios/Client/Frontend/Toolbar+URLBar/ShareButton.swift]: yes, missed it. Updated :)

**Review by lmarceau — APPROVED** [review]

**Comment by lmarceau** [comment]
Thank you @rajparad !

**Comment by rajparad** [comment]
> Thank you @rajparad !

Thanks for the help @lmarceau

**Pull request merged**